src/test/java/net/sf/jabref/logic/util/io/FileBasedTestCase.java

```diff
@@ -0,0 +1,124 @@
+package net.sf.jabref.logic.util.io;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import net.sf.jabref.BibtexTestData;
+import net.sf.jabref.Globals;
+import net.sf.jabref.model.database.BibDatabase;
+import net.sf.jabref.model.entry.BibEntry;
+import net.sf.jabref.preferences.JabRefPreferences;
+import net.sf.jabref.support.DevEnvironment;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assume.assumeFalse;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class FileBasedTestCase {
+
+    private BibEntry entry;
+    private Path rootDir;
+
+
+    @Before
+    public void setUp() throws IOException {
+        Globals.prefs = mock(JabRefPreferences.class);
+
+        BibDatabase database = BibtexTestData.getBibtexDatabase();
+        entry = database.getEntries().iterator().next();
+
+        rootDir = Files.createTempDirectory("UtilFindFileTest");
+
+
+
+        Path subDir = Files.createDirectory(rootDir.resolve("Organization Science"));
+        Path pdfSubDir = Files.createDirectory(rootDir.resolve("pdfs"));
+
+        assertTrue(Files.exists(Files.createFile(subDir.resolve("HipKro03 - Hello.pdf"))));
```

Comment:
I would prefer a smiple `Files.createFile` without the assertion for readability (and since I suspect that creating the file never fails...)